src/App.tsx

```diff
@@ -1,24 +1,257 @@
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterBy.ALL);
+  const [isError, setIsError] = useState<string | null>(null);
+  const [inputValue, setInputValue] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disableInput, setDisableInput] = useState(false);
+  const [loadingTodo, setLoadingTodo] = useState([0]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((loadedTodos: Todo[]) => {
+        setTodos(loadedTodos);
+      })
+      .catch((loadedError: Error) => {
+        setIsError(loadedError?.message ?? 'Error');
+      });
+  }, []);
+
+  useEffect(() => {
+    if (isError) {
+      setTimeout(() => {
+        setIsError(null);
+      }, 3000);
+    }
+  }, [isError]);
+
+  const addTodo = async (todoTitle: string) => {
+    try {
+      const newTodo = {
+        userId: USER_ID,
+        title: todoTitle,
+        completed: false,
+      };
+
+      setTempTodo({
+        id: 0,
+        ...newTodo,
+      });
+
+      const createdTodo = await postTodos(newTodo);
+
+      setTodos(currentTodos => [...currentTodos, createdTodo]);
+    } catch {
+      setIsError('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const updateTitle = async (
+    id: number,
+    title: string,
+  ) => {
+    try {
+      setLoadingTodo(prevTodosId => [...prevTodosId, id]);
+      const updatingTodo = todos.find(todo => todo.id === id);
+
+      if (!updatingTodo) {
+        return;
+      }
+
+      const hasTitle = title
+        ? { title }
+        : { completed: !updatingTodo?.completed };
+
+      const updatedTodo = {
+        ...updatingTodo,
+        ...hasTitle,
+      };
+
+      await updateTodoTitle(id, title);
+
+      setTodos(prevTodos => (
+        prevTodos.map(todo => (
+          todo.id !== id
+            ? todo
+            : updatedTodo
+        ))
+      ));
+    } catch {
+      setIsError('Unable to update a todo');
+    } finally {
+      setLoadingTodo(currentTodosId => currentTodosId
+        .filter(todoId => todoId !== id));
+    }
+  };
+
+  const updateStatus = async (id: number) => {
+    try {
+      setLoadingTodo(prevTodosId => [...prevTodosId, id]);
+      const updatingTodo = todos.find(todo => todo.id === id);
+
+      if (!updatingTodo) {
+        return;
+      }
+
+      await updateTodoStatus(id, !updatingTodo.completed);
+      setTodos(prevTodos => prevTodos.map(todo => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !todo.completed,
+          };
+        }
+
+        return todo;
+      }));
+    } catch {
+      setIsError('Unable to update a todo');
+    } finally {
+      setLoadingTodo(currentTodosId => currentTodosId
+        .filter(todoId => todoId !== id));
+    }
+  };
+
+  const handleFormSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
Which form do you submit here?
Move the event handler to the header component and provide only the `addTodo` function